### PR TITLE
GODRIVER-2827 Add prose test for change stream splitting

### PR DESCRIPTION
**Pull request by prestonvasquez**
GODRIVER-2827

## Summary

The Go Driver does not have strongly typed change stream events, this PR adds the specification-defined prose test to ensure this behavior.

## Background & Motivation

MongoDB 7.0 is adding a new aggregation stage $changeStreamSplitLargeEvent, which will split large events (>16MB) into smaller fragments.

### CODE REVIEW

**Review by blink1073 — APPROVED** [review]
LGTM!

File: mongo/integration/change_stream_test.go [hunk 1]
```diff
@@ -765,3 +853,13 @@ func compareResumeTokens(mt *mtest.T, cs *mongo.ChangeStream, expected bson.Raw)
...
+
+// generateLongString is a helper function to generate a long string of a
+// specific length
+func generateLongString(length int) string {
+	data := make([]byte, length)
+	for i := 0; i < length; i++ {
+		data[i] = 'q'
+	}
+	return string(data)
+}
```
Comment (qingyang-hu) [inline mongo/integration/change_stream_test.go:865]: We can simply use `strings.Repeat`.

Comment (prestonvasquez) [reply mongo/integration/change_stream_test.go]: Oh nice!

**Review by qingyang-hu — APPROVED** [review]
👍

**Pull request merged**